File: Programacion/Python/Ej_Recursividad/Ej8.py
def ackermann(m, n):
    if m == 0:
        return n+1
    elif m>0 and n == 0:
        return ackermann(m - 1, 1)
    else:
        return ackermann(m - 1, ackermann(m, n-1))

n1 = int(input("Ackermann, dime un número(1-3): "))
n2 = int(input("Ackermann, dime otro número (1-6): "))
print("Tu número ackerman es: ", ackermann(n1, n2))

File: DAM 3er_año_1º-2025-26/Programación/UD04 - Ej1/UD04 - EJ1/src/Ej8.java
/*
8. Crear una función que calcule el mayor de dos números enteros que recibirá
como parámetros. El resultado será otro número entero. Desarrolla otra
implementación que reciba 2 números reales y devuelva un número real
(double).
*/

import java.util.Scanner;

public class Ej8 {
    public static int mayorEntero(int n1, int n2) {
        return Math.max(n1, n2);
    }

    public static double mayorReal(double n1, double n2) {
        return Math.max(n1, n2);
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Dime un número entero: ");
        int n1 = scanner.nextInt();

        System.out.print("Dime otro número entero: ");
        int n2 = scanner.nextInt();

        System.out.println("Mayor de los enteros: " + mayorEntero(n1, n2));

        System.out.print("Dime un número real: ");
        double num1 = scanner.nextDouble();

        System.out.print("Dime otro número real: ");
        double num2 = scanner.nextDouble();

        System.out.println("Mayor de los reales: " + mayorReal(num1, num2));

    }
}
